Lib/cmd.py

```diff
@@ -245,8 +246,16 @@ def completedefault(self, *ignored):
         return []
 
     def completenames(self, text, *ignored):
+        """ Method returns all commands.
+
+        If hide_undoc is True, only the commands with "help_" are returned.
+        """
         dotext = 'do_'+text
-        return [a[3:] for a in self.get_names() if a.startswith(dotext)]
+        names = self.get_names()
+        commands = [a[3:] for a in names if a.startswith(dotext)]
+        if self.hide_undoc:
```

Comment:
Please add a test similar to https://github.com/python/cpython/blob/f1e17e9f97d9a4e97a5d99574775ee343a3a74fb/Lib/test/test_cmd.py#L55 . In the doctest you can set `mycmd.hide_undoc = True` and make sure "shell" is not returned by the `completenames` function for "s". Set it `mycmd.hide_undoc = False` for rest of the tests to make sure to make sure other tests are not affected.

Reply:
understood, but this is new to me, so i'll need a few days to figure this out and do this correctly

Reply:
ok, figured it out and made commit.